Automate anywhere config

Kyle-CW opened the pull request:
This change will automatically generate yaml config files that the sample game will use to read settings and connect to GameLift Anywhere.

This setup expects the plugin settings file `GameLiftSettings.yaml` to have been generated.

Using the menu buttons "Apply [Windows/MacOS] Sample [Client/Server] Build Settings" will automatically generate the files and place them in the project root. This will allow you to connect to Anywhere using the plugin settings within Unity Editor.
![image](https://github.com/aws/amazon-gamelift-plugin-unity/assets/109524503/edcde147-4591-4bd5-b9f5-db22db97267a)

I have also added `AnywhereFleetSettingsBuildProcessor` which will run after a build is complete annd will generate the config file in that build folder.


I also identified that while server settings keys were defined within the sample game project, client settings keys were defined within `Runtime`, so I have changed it so the sample game has the definitions and passes the values through to Runtime inside the existing configuration object

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

AWSJackson commented:
Summary from discussion during our sync today:
- We should update the SampleGame server code to pull settings directly from StateManager with the editor pre-processor so that a config file is not needed when running in the editor.
- When clicking Launch Server from the Anywhere plugin UI, it should update to the server pre-processor (probably a one-liner to call the same method used by the dropdown).
- We should investigate if it's possible to update the server vs client pre-processor property as part of switching between Dedicated Server and clients build targets in the Unity Build Manager.

Kyle-CW commented:
> When clicking Launch Server from the Anywhere plugin UI, it should update to the server pre-processor (probably a one-liner to call the same method used by the dropdown).

I have added functionality to the button to swap into server configuration and enter play mode

> We should investigate if it's possible to update the server vs client pre-processor property as part of switching between Dedicated Server and clients build targets in the Unity Build Manager.

I have added `BuildTargetChangedHandler.cs` which will receive a callback when the build target is changed, and add/remove the `UNITY_SERVER` symbol as appropriate.

> We should update the SampleGame server code to pull settings directly from StateManager with the editor pre-processor so that a config file is not needed when running in the editor.

So its a bit awkward with `StateManager` being in assembly `AmazonGameLiftPlugin.Editor` so we don't want to tie `SampleGame` assembly to an Editor assembly. I'll catch up with you later to decide on solutions to this.

AWSJackson commented on Editor/Window/AnywherePage.cs:
```diff
@@ -10,6 +11,8 @@ namespace AmazonGameLift.Editor
     {
         private readonly VisualElement _container;
+        private readonly ScriptingDefineSymbolEditor _defineEditor;
+        private const string _unityServerDefine = "UNITY_SERVER";
```
We have three separate constants for the same preprocessor symbol. We should move these someone constant (like ScriptingDefineSymbolEditor for now). If we add more symbols in the future it can be a separate file

AWSJackson commented on Editor/CoreAPI/GameLiftFleetManager.cs:
```diff
@@ -20,10 +20,11 @@ namespace Editor.CoreAPI
 {
     public class GameLiftFleetManager
     {
+        public readonly string FleetLocation = "custom-location-1";
```
Instead of making this a constant, we should instead return the location in the create fleet response

AWSJackson reviewed (changes requested):
I have quite a few comments, and this PR doesn't look like it would work for standalone builds or for managed ec2 builds. I'm going to apply some of my comments to try and merge this today and the rest can be done Monday next week.

AWSJackson commented on Editor/Window/ConnectToFleetInput.cs:
```diff
@@ -111,6 +111,7 @@ private void OnSelectFleetDropdown(string fleetName)
                 _fleetIdText.text = currentFleet.FleetId;
                 _stateManager.AnywhereFleetName = currentFleet.Name;
                 _stateManager.AnywhereFleetId = currentFleet.FleetId;
+                _stateManager.AnywhereFleetLocation = _fleetManager.FleetLocation;
```
I've added the same line of code to `OnAnywhereConnectClicked`, but see comment in GameLiftFleetManager about returning the location in the response instead of grabbing the constant

AWSJackson commented on Samples~/SampleGame/Assets/Scripts/GameLift.cs:
```diff
@@ -5,11 +5,13 @@
+using Editor.CoreAPI;
 #endif
 using UnityEngine;
 using UnityEngine.Events;
```
```suggestion
using UnityEngine.Events;
#if UNITY_EDITOR
using AmazonGameLift.Editor;
using Editor.CoreAPI;
#endif
```

AWSJackson commented on Samples~/SampleGame/Assets/Scripts/GameLift.cs:
```diff
@@ -5,11 +5,13 @@
 using System.Net;
 using System.Net.Security;
 using System.Security.Cryptography.X509Certificates;
+using AmazonGameLift.Editor;
```
```suggestion
```

AWSJackson reviewed (approved):
Approving this after my changes to unblock merging. A follow-up PR should be sent out to fix the straggler comments and make sure that building/running standalone executables is supported
